docs/architecture.md

```diff
@@ -27,14 +27,87 @@ Users requiring virtualization services are speaking to the Virtualization API
+3- Additional daemons for node specific logic associated with new types
+
+Once all three steps have been completed, you are able to
+
+- create new objects of these new types in Kubernetes (VMs in our
+  case)
+- and the new controllers take care to get the VMs scheduled on some host,
+- and a daemon - the `virt-handler` - is taking care on a host - alongside the
+  `kubelet` - to launch the VM and configure it until it matches the required
+  state.
+
+One a final note it is to say that both, the controllers and daemons are running
+as Pods (or similar) _on top of_ the Kubernetes cluster, and are not installed
+alongside of it. The type is - as said before - even defined inside the
+Kubernetes API server. This allows users to speak to Kubernetes, but modify VMs.
+
+The following diagram illustrates how the addditional controllers and daemons
+communicate with Kubernetes and where the additional types are stored:
+
+                             +-----------------+
+                             | virt-controller | (2)
+                             +-----------------+
+                                       |
+                                       | (watch for VMs, on cluster)
+                                       |
+                                       |
+                                       |   +----------------------+
+                                       |   |                      |
+                                       |   |                      |
+                                       v   v                      |
+                    +--------------+-----------+-----------+      |
+                    | Kubernetes   |  REST API |           |      |
+                    |              +-----------+           |      |
+                    |                                      |      |
+                    | [rc-controller] (1)                  |      |
+                    |                                      |      |
+                    | [Pod Foo] (1)                        |      |
+                    | [VM  Bar] (2)                        |      |
+                    +--------------------------------------+      |
+                                                                  |
+                                                                  |
+                                 +---------------------------+----+
+     (Cluster level)             |                           |
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+     (Node level)                |                           |
+                                 | (watch for VM changes,    |
+                                 |  on Node)                 |
+                                 |                           |
+                          +--------------+            +--------------+
+                          | virt-handler | (2)        | virt-handler | (2)
```

Comment:
Yeah, it should have one virt hander and 2 virt-launchers.  Maybe we should rename that to virt-stub before documenting?